### PR TITLE
Admin panel

### PR DESCRIPTION
**Pull request by lDarki**
1) Integrate map library
2) Home view
3) Device list table

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -22,8 +22,10 @@ module.exports = {
       }
     }
   },
+
   // add your custom rules here
   rules: {
+    "linebreak-style": ["error", "windows"],
```
Comment (Igogrek) [inline .eslintrc.js]: NO

File: index.html [hunk 2]
```diff
@@ -1,6 +1,9 @@
 <!DOCTYPE html>
 <html>
   <head>
+    <link rel="stylesheet" href="https://unpkg.com/leaflet@1.3.1/dist/leaflet.css" />
+    <link rel="stylesheet" href="https://unpkg.com/vue-material@beta/dist/vue-material.min.css">
```
Comment (Igogrek) [inline index.html]: Nope, use NPM package assets

File: package.json [hunk 3]
```diff
@@ -18,9 +18,13 @@
   },
   "dependencies": {
     "bulma": "^0.6.2",
+    "leaflet": "^1.3.1",
     "vue": "^2.5.2",
+    "vue-material": "^1.0.0-beta-7",
```
Comment (Igogrek) [inline package.json]: Let's not use this now

File: package.json [hunk 4]
```diff
@@ -18,9 +18,13 @@
...
     "vue-router": "^3.0.1",
-    "vuex": "^3.0.1"
+    "vue2-leaflet": "0.0.60",
```
Comment (Igogrek) [inline package.json]: Remove and use leaflet manually

File: package.json [hunk 5]
```diff
@@ -18,9 +18,13 @@
...
+    "vuex": "^3.0.1",
+    "axios": "latest"
```
Comment (Igogrek) [inline package.json]: exact version

File: src/components/common/box/box-component.vue [hunk 6]
```diff
@@ -0,0 +1,84 @@
+<template>
+  <div class="box">
+    <div class="box-header">
+      <span>{{title}}</span>
+    </div>
+    <section class="box-content" v-bind:class="[fillIn ? 'padding-of' : '']">
```
Comment (Igogrek) [inline src/components/common/box/box-component.vue]: use shorthand syntax : with {}

File: src/components/common/box/box-component.vue [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+  .box {
+    display: flex;
+    flex-direction: column;
+    flex: 1 1 100%;
+
+    color: $box-text-color;
+    background: white;
+
+    box-sizing: border-box;
+    box-shadow: 0 1px 1px 0.003em rgba(0, 0, 0, .16);
```
Comment (Igogrek) [inline src/components/common/box/box-component.vue]: 0.003em?

File: src/components/device/home/device-panel.vue [hunk 8]
```diff
@@ -0,0 +1,82 @@
+<template>
+  <section class="device-panel">
+    <box-component class="alarm-box" title="ACTIVE, CRITICAL ALARMS">
+      <div class="box-row">
+        <img class="icon" src="../../assets/alert-idle.svg"/>
+        <span>No alarms to display</span>
+      </div>
+
+    </box-component>
+    <box-component class="map-box" title="MAP" v-bind:fillIn="true">
```
Comment (Igogrek) [inline src/components/device/home/device-panel.vue]: again, shorthand syntax

File: src/components/device/home/device-panel.vue [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+      <div id="app" style="height: 100%">
```
Comment (Igogrek) [inline src/components/device/home/device-panel.vue]: id app? Remove. Also don't use style attribute

File: src/components/device/home/device-panel.vue [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+        <v-map :zoom=13 :center="[47.413220, -1.219482]">
+          <v-tilelayer url="http://{s}.tile.osm.org/{z}/{x}/{y}.png"></v-tilelayer>
+          <v-marker :lat-lng="[47.413220, -1.219482]"></v-marker>
```
Comment (Igogrek) [inline src/components/device/home/device-panel.vue]: Please extract constants

File: src/components/device/home/device-panel.vue [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+    <box-component class="alarm-box" title="ACTIVE, CRITICAL ALARMS">
+      <div class="box-row">
+        <img class="icon" src="../../assets/alert-idle.svg"/>
+        <span>No alarms to display</span>
+      </div>
+
+    </box-component>
+    <box-component class="map-box" title="MAP" v-bind:fillIn="true">
+      <div id="app" style="height: 100%">
+        <v-map :zoom=13 :center="[47.413220, -1.219482]">
```
Comment (Igogrek) [inline src/components/device/home/device-panel.vue]: Just use leaflet

File: src/components/device/home/device-panel.vue [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+</template>
+
+
+<script lang="ts">
+
+  import Vue from 'vue'
+  import BoxComponent from '../../common/box/box-component.vue'
+
+  declare const require: (moduleId: string) => any;
```
Comment (Igogrek) [inline src/components/device/home/device-panel.vue]: What? Remove

File: src/components/device/home/device-panel.vue [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+  const Vue2Leaflet = require("vue2-leaflet");
```
Comment (Igogrek) [inline src/components/device/home/device-panel.vue]: use ES6 import

File: src/components/device/list/device-list.vue [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+</script>
+
+<style lang="scss" scoped>
+  .list {
+    display: flex;
+    flex-direction: column;
+    flex: 1 1 100%;
+    background-color: white;
+    color: black;
+    box-shadow: 0 1px 1px 0.003em rgba(0, 0, 0, .16);
```
Comment (Igogrek) [inline src/components/device/list/device-list.vue]: Same shadow - extract style

File: src/store/modules/devices/mutations.ts [hunk 15]
```diff
@@ -0,0 +1,29 @@
+import {SET_DEVICES} from "../types/device-types";
+import {DeviceState} from "./index";
+import {ConnectionStatus, DeviceListNode} from "../../../components/dto/DeviceListNode";
+
+export const mutations = {
+  [SET_DEVICES](state: DeviceState, devices: Object) {
+    debugger;
```
Comment (Igogrek) [inline src/store/modules/devices/mutations.ts]: remove

File: src/store/modules/devices/mutations.ts [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+    state.devices = mapObjToDevices(devices);
+  }
+};
+
+function mapObjToDevices(devices: any): Array<DeviceListNode> {
+  return devices.filter(obj => obj.name !== undefined)
```
Comment (Igogrek) [inline src/store/modules/devices/mutations.ts]: Just do Boolean(obj.name)

File: src/components/device/list/device-list.vue (hunk 14 above)
Comment (lDarki) [reply src/components/device/list/device-list.vue]: fixed

File: src/store/modules/devices/mutations.ts (hunk 16 above)
Comment (lDarki) [reply src/store/modules/devices/mutations.ts]: fixed

File: src/store/modules/devices/mutations.ts (hunk 15 above)
Comment (lDarki) [reply src/store/modules/devices/mutations.ts]: fixed

**Pull request merged**